pkg/util/grpcencoding/snappy/snappy.go

```diff
@@ -85,3 +85,17 @@ func (r reader) Read(p []byte) (n int, err error) {
 	}
 	return n, err
 }
+
+// If a Compressor implements DecompressedSize(compressedBytes []byte) int,
+// gRPC will call it to determine the size of the buffer allocated for the
+// result of decompression.
+// Return -1 to indicate unknown size.
+//
+// This is an EXPERIMENTAL feature of grpc-go.
+func (r reader) DecompressedSize(compressedBytes []byte) int {
```

Comment:
humm

This should not be in the `compressor` struct?

https://github.com/grpc/grpc-go/blob/a1e657ce53ba59e6e4076edaa2c68d398e7f7e4f/rpc_util.go#L765-L789

Reply:
O.. My fault.. https://github.com/grpc/grpc-go/blob/6f44ae89b1ab4e492516fd8f1770ad16451a9f02/encoding/encoding.go#L39-L63

Reply:
CPU usage comparison, roughly 25% decrement on average
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/6223594/225532273-59afe8c8-b64e-4eb5-9acc-fc017afd6c18.png">